### PR TITLE
[ts] Update network-facing libs like express (et al.), node-fetch, p-…

### PR DESCRIPTION
**Pull request by geropl**
…throttle, probot

## Description

Besides the update, this PR sets explicit timeouts on all `fetch` calls. Hope this did not cause our increasing lag: firewalls/rate-limiter for github.com `DROP` ing our packets might very well have caused this behavior. :see_no_evil: 

This took a bit longer than expected: We have (by design) divergence how our tests are executed (leeway vs yarn test), and that broke here. 

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: components/server/src/init.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ process.on("SIGUSR1", () => {
                 if (!err) {
                     const filename = path.join(os.tmpdir(), Date.now() + '.cpuprofile');
                     console.log('preparing cpuprofile: ' + filename);
-                    fs.promises.writeFile(filename, JSON.stringify(profile));
+                    fs.promises.writeFile(filename, JSON.stringify(profile)).catch(err => console.error("error writing cpuprofile", err));
```
Comment (JanKoehnlein) [inline components/server/src/init.ts:80]: See also https://github.com/gitpod-io/gitpod/pull/8306/files#:~:text=components/server/src/init.ts

**Review by JanKoehnlein — DISMISSED** [review]
LGTM.

File: components/server/src/init.ts (hunk 1 above)
Comment (AlexTugarev) [reply components/server/src/init.ts:80]: Thanks!

**Comment by geropl** [comment]
/werft run

:crossed_fingers: that certs work now

**Comment by geropl** [comment]
Finally green - sorry for the noise here.

File: components/ee/payment-endpoint/package.json [hunk 2]
```diff
@@ -9,12 +9,12 @@
...
     "@octokit/webhooks": "9.17.0",
-    "body-parser": "^1.19.0",
+    "body-parser": "^1.19.2",
     "chargebee": "^2.4.6",
-    "express": "^4.17.1",
+    "express": "^4.17.3",
     "inversify": "^5.0.1",
     "jsonwebtoken": "^8.5.0",
-    "node-fetch": "^2.6.1",
+    "node-fetch": "^2.6.7",
```
[the remark this answers is not in the document]
Comment (geropl) [reply components/ee/payment-endpoint/package.json:17]: > why not bumping to 3.2.0?

Tried that first, but landed in the "import as ESM module" rabbithole. Digged a bit, but eventually backed off. :roll_eyes:

[the remark this answers is not in the document]
Comment (geropl) [reply components/ee/payment-endpoint/package.json:17]: `2.6.7` has the same security patches as `3.2.0` :+1:

File: components/server/package.json [hunk 3]
```diff
@@ -65,16 +65,16 @@
     "lodash.debounce": "^4.0.8",
     "longjohn": "^0.2.12",
     "minio": "^7.0.12",
-    "node-fetch": "^2.6.1",
+    "node-fetch": "^2.6.7",
     "opentracing": "^0.14.4",
-    "p-throttle": "^3.0.0",
+    "p-throttle": "^5.0.0",
```
Comment (AlexTugarev) [inline components/server/package.json:70]: > require() of ES Module /app/node_modules/p-throttle/index.js

same issue

Comment (AlexTugarev) [reply components/server/package.json:70]: `"type": "module"` in package.json?

Comment (geropl) [reply components/server/package.json:70]: >>    require() of ES Module /app/node_modules/p-throttle/index.js

> same issue

What did you do? And why did the build run? :grimacing:

Comment (AlexTugarev) [reply components/server/package.json:70]: server is crashing in the preview deployment

Comment (geropl) [reply components/server/package.json:70]: > "type": "module" in package.json?

Not, it's not that easy. Maybe not too complicated either, but I want the other changes to be in by today. I will remove pthrottle, then.

Can you give be your cmdline to reproduce?

**Review by AlexTugarev — APPROVED** [review]
Changes look good! Thanks

(image pull issues in preview env are not related.)

**Pull request merged**